recommender/conf/init.conf

```diff
@@ -0,0 +1,33 @@
+# Upstart service configuration
+start on runlevel [2345]
+stop on runlevel [016]
+
+env APP_NAME=recommender
+env USER_HOME=/recommender-1.0-SNAPSHOT
+
+setuid recommender
+chdir /recommender-1.0-SNAPSHOT
+
+script
+LOGFILE=$USER_HOME/stdout.log
+
+TOTAL_MEMORY=$(grep MemTotal /proc/meminfo | awk '{ print $2 }')
+HEAP_SIZE_IN_MB=$(python -c "print int($TOTAL_MEMORY * 0.8 / 1024)")
+
+JAVA_GC="-J-XX:NewRatio=1 \
+-J-XX:+PrintGCDetails \
+-J-XX:+PrintGCDateStamps \
+-J-XX:+UseCompressedOops \
+-J-XX:+UseConcMarkSweepGC \
+-J-XX:+CMSClassUnloadingEnabled \
+-J-verbose:gc \
+-J-Xloggc:$USER_HOME/gc.log \
+-J-XX:+UseGCLogFileRotation \
+-J-XX:NumberOfGCLogFiles=5 \
+-J-XX:GCLogFileSize=5M"
+
+echo "$COMMAND" > $USER_HOME/cmd.txt
+
+COMMAND="$USER_HOME/bin/$APP_NAME -J-Xmx${HEAP_SIZE_IN_MB}m -J-Xms${HEAP_SIZE_IN_MB}m $JAVA_GC"
```

Comment:
This instance will be running two java apps, this one and "pio deploy".  Should probably reduce the maximum heap size to allow for this